opentelemetry-otlp/tests/integration_test/tests/integration_tests.rs

```diff
@@ -0,0 +1,93 @@
+use integration_test_runner::images::Collector;
+use std::fs::File;
+use std::os::unix::fs::PermissionsExt;
+use std::time::Duration;
+use testcontainers::clients::Cli;
+use testcontainers::core::Port;
+use testcontainers::RunnableImage;
+
+mod traces;
+
+const COLLECTOR_CONTAINER_NAME: &str = "otel-collector";
+const TEST_RESULT_DIR_IN_CONTAINER: &str = "testresults";
+const EXPECTED_DIR: &str = "./expected";
+
+struct TestSuite {
+    result_file_path: &'static str,
+}
+
+impl TestSuite {
+    fn new(result_file_path: &'static str) -> Self {
+        Self { result_file_path }
+    }
+
+    pub fn expected_file_path(&self) -> String {
+        format!("{}/{}", EXPECTED_DIR, self.result_file_path)
+    }
+
+    pub fn result_file_path_in_container(&self) -> String {
+        format!(
+            "/{}/{}",
+            TEST_RESULT_DIR_IN_CONTAINER, self.result_file_path
+        )
+    }
+
+    pub fn result_file_path(&self) -> String {
+        format!("./{}", self.result_file_path)
+    }
+
+    /// Create a empty file on localhost and copy it to container with proper permissions
+    /// we have to create the file for the container otherwise we will encounter a permission denied error.
+    /// see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3159
+    pub fn create_temporary_result_file(&self) -> File {
+        let file = File::create(self.result_file_path()).unwrap();
+        file.set_permissions(std::fs::Permissions::from_mode(0o666))
+            .unwrap();
+        file
+    }
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[ignore] // skip when running unit test
```

Comment:
do we only expect to run this manually? Or we expect this to be run in CI as well?

Reply:
nvm. found that we run even ignored in intergration tests.